lib/mongo/uri.rb

```diff
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-require 'uri'
+require 'addressable'
```

Comment:
I found that Rack is using 

```ruby
::URI::DEFAULT_PARSER.unescape
```
in its unescape_path method:
http://www.rubydoc.info/github/rack/rack/Rack%2FUtils.unescape_path

so I'll do the same.